node.go

```diff
@@ -208,6 +228,8 @@ func (n *ProtoNode) GetLinkedNode(ctx context.Context, ds format.DAGService, nam
 // NOTE: Does not make copies of Node objects in the links.
 func (n *ProtoNode) Copy() format.Node {
 	nnode := new(ProtoNode)
+	n.mu.RLock()
```

Comment:
might as well use defers consistently? one or two static defers are basically free now, as the compiler is efficient with them

Reply:
My understanding was that defer()s are pretty expensive, will change...

Reply:
See https://github.com/golang/go/issues/14939. That was the case until 2019 :)